### PR TITLE
Initial work on zone based location tracking.

### PR DESCRIPTION
**Pull request by JBassett**

### CODE REVIEW

File: domain/src/main/java/io/homeassistant/companion/android/domain/integration/EntityResponse.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package io.homeassistant.companion.android.domain.integration
+
+import java.util.Calendar
+
+data class EntityResponse(
+    val entityId: String,
+    val state: String,
+    val attributes: Map<String, String>,
```
Comment (balloob) [inline domain/src/main/java/io/homeassistant/companion/android/domain/integration/EntityResponse.kt]: This is `Map<String, Any>`

Comment (balloob) [reply domain/src/main/java/io/homeassistant/companion/android/domain/integration/EntityResponse.kt]: If it helps, in TypeScript we have a base entity and then `ZoneEntity`, `LightEntity` etc where we fill in the attributes with the types we expect.

File: domain/src/main/java/io/homeassistant/companion/android/domain/integration/EntityResponse.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    val lastChanged: Calendar,
+    val lastUpdated: Calendar,
+    val context: Map<String, String>
```
Comment (balloob) [inline domain/src/main/java/io/homeassistant/companion/android/domain/integration/EntityResponse.kt]: This also is `Any` for values

**Comment by balloob** [comment]
This looks good on the surface. I don't know enough to judge if enough tests / background work is good. So a 👍 from me.

**Comment by robbiet480** [comment]
Closing in favor of just merged #129

**Pull request closed without merging**